fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/DefaultApplicationProvider.java

```diff
@@ -85,6 +87,24 @@ public DefaultApplicationProvider(Front50Service front50Service,
     }
   }
 
+  @Override
+  public Set<Application> addItem(Application application) {
+    Set<Application> applicationSet = getAll();
+    if (applicationSet == null) {
+      log.error("The application cache is empty. (This is ok if you do not have any applications at all.)");
```

Comment:
If this could be ok, I'd make it `log.warn`

Reply:
For all cases where you have an app or more in spinnaker this is a case you should not reach. So I think it is an error. This should only appear when you do not have any apps at all and you create the first one and the cache is younger than 20 seconds.